src/ng/directive/input.js

```diff
@@ -1290,6 +1290,9 @@ function createDateParser(regexp, mapping) {
           }
         });
         return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+      } else {
+        // If we fail to parse the string with the ISO regex, it might be because it's the result of Date.toString()
+        return Date.parse(iso);
```

Comment:
I think that the 6754 and 6755 closes were just the wrong way round.

Changing the failing tests to use `toISOString()` prevents the hack in `input.js`.

The question is whether we need additional tests that specifically check that we can use real `Date` objects as well?